vpr/src/device/rr_graph_view.h

```diff
@@ -379,6 +379,14 @@ class RRGraphView {
     inline const t_segment_inf& rr_segments(RRSegmentId seg_id) const {
         return rr_segments_[seg_id];
     }
+    inline size_t num_rr_segments() const {
+        return rr_segments_.size();
+    }
+    /** @brief Return the rr_segments structure for queries from client functions */
```

Comment:
Slight modification: 

``Return a read-only list of rr_segments for queries from client functions ``